packages/comms/src/ecl/query.ts

```diff
@@ -76,41 +74,81 @@ export class Query extends StateObject<QueryEx, QueryEx> implements QueryEx {
         return retVal;
     }
 
+
+    private _eclService: Promise<EclService>;
+    protected async wsEclService(): Promise<EclService | undefined> {
+        if (!this._eclService) {
+            this._eclService = this.topology.fetchServices({}).then(services => {
+                for (const espServer of services?.TpEspServers?.TpEspServer) {
```

Comment:
These couple for...of loops were the only thing I saw as possible problems, and GH actions already made annotations. Optionally chained forEach()s instead maybe? Though is it possible for the ESP responses to even have those as undefined, rather than empty arrays? I don't think I've ever run into that at least.